lib/rules/no-mixed-spaces-and-tabs.js

```diff
@@ -41,72 +40,30 @@ module.exports = {
                 smartTabs = false;
         }
 
-        /**
-         * Determines if a given line and column are before a location.
-         * @param {Location} loc The location object from an AST node.
-         * @param {int} line The line to check.
-         * @param {int} column The column to check.
-         * @returns {boolean} True if the line and column are before the location, false if not.
-         * @private
-         */
-        function beforeLoc(loc, line, column) {
-            if (line < loc.start.line) {
-                return true;
-            }
-            return line === loc.start.line && column < loc.start.column;
-        }
-
-        /**
-         * Determines if a given line and column are after a location.
-         * @param {Location} loc The location object from an AST node.
-         * @param {int} line The line to check.
-         * @param {int} column The column to check.
-         * @returns {boolean} True if the line and column are after the location, false if not.
-         * @private
-         */
-        function afterLoc(loc, line, column) {
-            if (line > loc.end.line) {
-                return true;
-            }
-            return line === loc.end.line && column > loc.end.column;
-        }
-
         //--------------------------------------------------------------------------
         // Public
         //--------------------------------------------------------------------------
 
         return {
 
-            TemplateElement(node) {
-                ignoredLocs.push(node.loc);
-            },
-
             "Program:exit"(node) {
+                const lines = sourceCode.lines,
+                    comments = sourceCode.getAllComments(),
+                    ignoredCommentLines = new Set();
+
+                // Add all lines except the first ones.
+                comments.forEach(comment => {
```

Comment:
Maybe I'm missing something, but will the rule ignore lines like the following now?

```js
\t  const a /* comment */ = 1;
\t  const b = 1; // comment
```

Reply:
It shouldn't happen because the `for` loop inside is supposed to skip the first line, but it's certainly a good thing to add tests to confirm this.

Reply:
Added more tests for comments now.